app/lib/services/background_service.dart

```diff
@@ -10,14 +15,28 @@ void backgroundFetchHeadlessTask(HeadlessTask task) async {
     BackgroundFetch.finish(taskId);
     return;
   }
-  await checkNodeStatus();
+  final prefs = await SharedPreferences.getInstance();
+  final bool notificationsEnabled =
+      prefs.getBool(_nodeStatusNotificationEnabledKey) ?? true;
 
-  BackgroundFetch.finish(taskId);
+  logger.i(
+      'Background Fetch Headless Task: $taskId, Notifications Enabled: $notificationsEnabled');
+
+  if (!notificationsEnabled) {
+    logger.i(
+        '[BackgroundFetch] Node status notifications are disabled. Finishing task: $taskId');
+    BackgroundFetch.finish(taskId);
+    return;
+  }
+  await checkNodeStatus(taskId);
 }
 
-Future<void> checkNodeStatus() async {
+Future<void> checkNodeStatus(String taskId) async {
```

Comment:
As `BackgroundFetch.finish(taskId);` is called multiple times, it can be added only in `finally` block.